apps/backend/src/websocket/connection_manager.py

```
@@ -0,0 +1,249 @@
+        self.authenticated = False
+        self.user_id: Optional[str] = None
+        self.last_heartbeat = asyncio.get_event_loop().time()
+        
+    async def send_message(self, message: WebSocketMessage):
+        """Send a message to this connection"""
+        try:
+            # Check if WebSocket is still open before sending
+            if self.websocket.client_state.name != "CONNECTED":
+                logger.debug(f"WebSocket {self.client_id} not connected, skipping message")
+                return
+                
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)